fluss-client/src/main/java/com/alibaba/fluss/client/write/Sender.java

```diff
@@ -329,25 +329,38 @@ private void sendWriteRequest(int destination, short acks, List<WriteBatch> batc
                             .add(batch);
                 });
 
-        TabletServerGateway gateway = metadataUpdater.newTabletServerClientForNode(destination);
-        writeBatchByTable.forEach(
-                (tableId, writeBatches) -> {
-                    TableInfo tableInfo = metadataUpdater.getTableInfoOrElseThrow(tableId);
-                    if (tableInfo.hasPrimaryKey()) {
-                        sendPutKvRequestAndHandleResponse(
-                                gateway,
-                                makePutKvRequest(tableId, acks, maxRequestTimeoutMs, writeBatches),
-                                tableId,
-                                recordsByBucket);
-                    } else {
-                        sendProduceLogRequestAndHandleResponse(
-                                gateway,
-                                makeProduceLogRequest(
-                                        tableId, acks, maxRequestTimeoutMs, writeBatches),
-                                tableId,
-                                recordsByBucket);
-                    }
-                });
+        ServerNode destinationNode = metadataUpdater.getTabletServer(destination);
+        if (destinationNode == null) {
+            LOG.warn(
+                    "Server {} is not found in metadata cache, "
+                            + "going to request metadata update.",
+                    destination);
+            // invalidate the metadata to request metadata
+            invalidBucketMetadata(batches);
+            // requeue the write batch
+            batches.forEach(this::reEnqueueBatch);
```

Comment:
We can unify the logic with `handleWriteRequestException`. The batch re-enqueue is more complex than just calling `reEnqueueBatch`.